api/nodes/serializers.py

```diff
@@ -266,6 +269,111 @@ def update(self, node, validated_data):
         return node
 
 
+class NodeAddonSettingsSerializer(JSONAPISerializer):
+    """
+    Overrides AddonSettingsSerializer to return node-specific fields
+    """
+    class Meta:
+        type_ = 'node_addons'
+
+    id = ser.CharField(source='_id', read_only=True)
+    enabled = ser.BooleanField(required=True)
+    auth_id = ser.CharField(source='object.external_account._id', allow_null=True)
+    folder_id = ser.CharField(source='object.folder_id', allow_null=True)
+    folder_path = ser.CharField(source='object.folder_path', required=False, allow_null=True)
+    has_auth = ser.BooleanField(source='object.has_auth', read_only=True)
+    complete = ser.BooleanField(source='object.complete', read_only=True)
+    configured = ser.BooleanField(source='object.configured', read_only=True)
+    node = ser.CharField(source='object.owner._id', read_only=True)
+
+    def to_representation(self, data, envelope='data'):
+        return super(NodeAddonSettingsSerializer, self).to_representation(data, envelope=envelope)
+
+    def update(self, instance, validated_data):
+        addon_name = instance.get('_id', None)
+        if addon_name not in ADDONS_FOLDER_CONFIGURABLE:
+            raise exceptions.MethodNotAllowed('Requested addon not currently configurable via API.')
+
+        auth = get_user_auth(self.context['request'])
+        node_settings = instance.get('object', None)
+
+        try:
+            auth_id = validated_data['object']['external_account']['_id']
```

Comment:
`instance` is a dict passed in from the view, grabbed from the database, which contains the node settings object.
`validated_data` is a dict passed in from the user's request, parsed and validated by DRF, which contains data that the user wants to set on the node settings object.

You could maybe get an `external_account_id` from `node_settings`, but it wouldn't be one the user requested to connect, unless it was already connected.